File: chrome-extension/icons/generate-icons.py
#!/usr/bin/env python3
"""Generate Fuse extension icons in PNG format."""

try:
    from PIL import Image, ImageDraw, ImageFont
    import os
except ImportError:
    print("Error: PIL/Pillow not found. Install it with:")
    print("  pip install Pillow")
    print("\nAlternatively, open create-icons.html in a browser and save the icons manually.")
    exit(1)

def create_icon(size, filename):
    """Create a simple icon with 'F' on black background."""
    # Create a black background
    img = Image.new('RGBA', (size, size), color='black')
    draw = ImageDraw.Draw(img)
    
    # Calculate font size (70% of icon size)
    font_size = int(size * 0.7)
    
    try:
        # Try to use a system font
        font = ImageFont.truetype("Arial", font_size)
    except:
        try:
            font = ImageFont.truetype("/System/Library/Fonts/Helvetica.ttc", font_size)
        except:
            # Fallback to default font
            font = ImageFont.load_default()
    
    # Draw the 'F' text in white, centered
    text = "F"
    
    # Get text bounding box for centering
    bbox = draw.textbbox((0, 0), text, font=font)
    text_width = bbox[2] - bbox[0]
    text_height = bbox[3] - bbox[1]
    
    x = (size - text_width) // 2 - bbox[0]
    y = (size - text_height) // 2 - bbox[1]
    
    draw.text((x, y), text, fill='white', font=font)
    
    # Save the image
    img.save(filename)
    print(f"✓ Created {filename}")

# Get the directory where this script is located
script_dir = os.path.dirname(os.path.abspath(__file__))

# Generate all three icon sizes
sizes = [16, 48, 128]
for size in sizes:
    filename = os.path.join(script_dir, f'icon{size}.png')
    create_icon(size, filename)

print("\n✓ All icons generated successfully!")
print("You can now load the extension in Chrome.")

File: chrome-extension/icons/generate-icons.js
const fs = require('fs');
const path = require('path');

// Simple function to create a basic PNG icon with "F" text
function createIconSVG(size) {
  return `<?xml version="1.0" encoding="UTF-8"?>
<svg width="${size}" height="${size}" xmlns="http://www.w3.org/2000/svg">
  <rect width="${size}" height="${size}" fill="#000000" rx="${size * 0.0625}"/>
  <text x="50%" y="50%" font-family="Arial, sans-serif" font-size="${size * 0.7}" font-weight="bold" fill="#ffffff" text-anchor="middle" dominant-baseline="central">F</text>
</svg>`;
}

// Create SVG files first
const sizes = [16, 48, 128];
const iconsDir = __dirname;

sizes.forEach(size => {
  const svgContent = createIconSVG(size);
  const svgPath = path.join(iconsDir, `icon${size}.svg`);
  fs.writeFileSync(svgPath, svgContent);
  console.log(`Created ${svgPath}`);
});

console.log('\nSVG icons created! To convert to PNG, you can:');
console.log('1. Use an online converter like https://cloudconvert.com/svg-to-png');
console.log('2. Use ImageMagick: convert icon16.svg icon16.png');
console.log('3. Use the create-icons.html file and save manually');
console.log('\nOr install sharp for automatic conversion: npm install sharp');
console.log('Then run: node convert-to-png.js');




